lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -309,47 +309,47 @@ static void MarkUsedSignatureElements(Function *F, DxilModule &DM) {
         continue;
       if (!GetUnsignedVal(LI.get_rowIndex(), &row))
         bDynIdx = true;
-      pSig = &DM.GetInputSignature();
+      pSig = &DM.GetDxilEntryProps(F).sig.InputSignature;
```

Comment:
It would be better to either pass the entry props for this entry into this function, or to look it up once outside the loop.